src/Context/video-context.js

```diff
@@ -0,0 +1,37 @@
+import axios from "axios";
+import { createContext,useContext, useEffect,useState,useReducer} from "react";
+import { getFilteredByCategory,getFilteredBySearchQuery } from "./Reducer/filterUtils";
+import { filterProductReducer } from "./Reducer/filterProductReducer";
+
+
+const initialState={
+    currentCategory:"All",
+    searchByQuery:""
+}
+
+const VideoContext =createContext();
+const VideoProvider =({children})=>{
+    const [videos,setVideos]=useState([]);
+    const [state,dispatch]=useReducer(filterProductReducer,initialState);
+    useEffect(()=> {
+        (async ()=>{
+            try{
```

Comment:
why 2times arrow functions ?

Reply:
no particular reason. i have written like this from starting. Is it a bad practice?

Reply:
I'm not sure but it gave errors to me sometimes. It's fine ig if it works.